### PR TITLE
Create script to create backups of all databases

### PR DESCRIPTION
**Pull request by cbickel**
Signed-off-by: Markus Thoemmes <markus.thoemmes@de.ibm.com>

### CODE REVIEW

File: tools/db/replicateDbs.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2015-2016 IBM Corporation
```
Comment (rabbah) [inline tools/db/replicateDbs.py]: 17? per your other pr.

Comment (cbickel) [inline tools/db/replicateDbs.py]: If the other PR is not merged, scanCode will fail if I use 2017 here.

**Comment by jasonpet** [comment]
I do not have access rights to merge into open.  You will need to reassign this.

File: tests/src/whisk/core/database/test/ReplicatorTests.scala [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    /** Compares to databases to full equality */
+    def compareDatabases(dbNames: Seq[String]) = {
+        val dbDocuments = dbNames.map { dbName =>
+            val client = new ExtendedCouchDbRestClient(config.dbProtocol, config.dbHost, config.dbPort.toInt, config.dbUsername, config.dbPassword, dbName)
+            val documents = client.getAllDocs(includeDocs = Some(true)).futureValue
+            documents shouldBe 'right
+            documents.right.get
+        }
+
+        all(dbDocuments.tail) shouldBe dbDocuments.head
```
Comment (jeremiaswerner) [inline tests/src/whisk/core/database/test/ReplicatorTests.scala]: nice

**Comment by jeremiaswerner** [comment]
very nice. LGTM

**Comment by rabbah** [comment]
Why is the pip install of couchdb needed?

**Comment by markusthoemmes** [comment]
> Why is this needed now? Is it compatible with cloudant?

Needed to successfully operate OpenWhisk :). Yes its fully compatible with either CouchDB or Cloudant.

> Why is the pip install of couchdb needed?

It's needed to be able to run the script (it depends on the couchdb client).

**Comment by markusthoemmes** [comment]
PG2 979

File: tests/src/whisk/core/database/test/ReplicatorTests.scala [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+    }
+
+    /** Wait for a document to appear */
+    def waitForDocument(client: ExtendedCouchDbRestClient, id: String) = waitfor(() => client.getDoc(id).futureValue.isRight)
+
+    /** Compares to databases to full equality */
+    def compareDatabases(dbNames: Seq[String]) = {
+        val dbDocuments = dbNames.map { dbName =>
+            val client = new ExtendedCouchDbRestClient(config.dbProtocol, config.dbHost, config.dbPort.toInt, config.dbUsername, config.dbPassword, dbName)
+            val documents = client.getAllDocs(includeDocs = Some(true)).futureValue
```
Comment (rabbah) [inline tests/src/whisk/core/database/test/ReplicatorTests.scala]: should put a comment somewhere here that this should not be used to test large databases.

File: tests/src/whisk/core/database/test/ReplicatorTests.scala [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+    behavior of "Database replication script"
+
+    it should "replicate a database" in {
+        // Create a database to backup
+        val dbName = dbPrefix + "backup_database"
+        val client = createDatabase(dbName)
+
+        println(s"Creating testdocument")
+        val testDocument = JsObject("malex" -> "markus+alex".toJson)
```
Comment (rabbah) [inline tests/src/whisk/core/database/test/ReplicatorTests.scala]: you sneaky boy you.

**Comment by jeremiaswerner** [comment]
rebased and merged in 7f998a948c3a0b4a3f62f09cb059e83290bc642c

**Pull request closed without merging**